x-pack/legacy/plugins/canvas/public/components/editor/editor_theme.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api';
+
+import darkTheme from '@elastic/eui/dist/eui_theme_dark.json';
+import lightTheme from '@elastic/eui/dist/eui_theme_light.json';
+
+import chrome from 'ui/chrome';
+
+const IS_DARK_THEME = chrome.getUiSettingsClient().get('theme:darkMode');
+
+const themeName = IS_DARK_THEME ? darkTheme : lightTheme;
+
+const themeColors = {
+  keyword: themeName.euiColorAccent,
+  comment: themeName.euiColorDarkShade,
+  delimiter: themeName.euiColorSecondary,
+  string: themeName.euiColorPrimary,
+  number: themeName.euiColorWarning,
+  regexp: themeName.euiColorPrimary,
+  types: `${IS_DARK_THEME ? themeName.euiColorVis5 : themeName.euiColorVis9}`,
+  annotation: themeName.euiColorLightShade,
+  tag: themeName.euiColorAccent,
+  symbol: themeName.euiColorDanger,
+  foreground: themeName.euiColorDarkestShade,
+  editorBackground: themeName.euiColorEmptyShade,
+  lineNumbers: themeName.euiColorDarkShade,
+  editorIndentGuide: themeName.euiColorLightShade,
+  selectionBackground: `${IS_DARK_THEME ? '#343551' : '#E3E4ED'}`,
+  editorWidgetBackground: themeName.euiColorLightestShade,
+  editorWidgetBorder: themeName.euiColorLightShade,
+  findMatchBackground: themeName.euiColorWarning,
+  findMatchHighlightBackground: themeName.euiColorWarning,
+};
+
+export const theme: monacoEditor.editor.IStandaloneThemeData = {
```

Comment:
Can we find a way to get this working with CodeSearch's implementation? What differs? I know we're building this one fast, but it seems like something we should be doing rather than having two separate implementations out there.

https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/code/public/monaco/monaco.ts

Reply:
Yeah, I forked over there theming work they had in place. I'm not sure this implementation of Monaco stuff will be re-used over by the Code Search team since their work focuses on read-only Monaco with a decent amount of adjustments.

I figured that the first version of this Editor component that is Canvas specific can use the separate "forked" version of the theming and then when we pull this Editor up to be usable by all of Kibana, we can attempt to unify the theme work done by Code team.

That was my gameplan but I'm open to discussion if that isn't what we want to do

Reply:
I added this as a line item for discussion here as well https://github.com/elastic/kibana/issues/43814